### PR TITLE
Changes introduced by work on academic paper

### PR DESCRIPTION
**Pull request by PBia**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,13 +1,40 @@
...
-    description="Hfinger - fingerprinting HTTP requests stored in pcap files",
+    version="0.2.0",
+    description="Hfinger - fingerprinting malware HTTP requests stored in pcap files",
     author="Piotr Bialczak",
     author_email="piotrb@cert.pl",
-    packages=["hfinger", "hfinger.configs"],
-    package_data={"hfinger": ["configs/*.json", "configs/*.txt"]},
-    install_requires=["fnvhash", "python-magic"],
+    packages=["hfinger"],
+    license="GPLv3",
```
Comment (chivay) [inline setup.py]: license kwarg is meant to be used when none of the classifiers is appropriate (thanks, @psrok1 )

File: README.md [hunk 2]
```diff
@@ -1,108 +1,180 @@
...
-It's main objective is to provide a representation of malware requests in a shorter form than printing whole 
-request, but still human interpretable. This representation should be unique between malware families, 
-what means that any fingerprint should be seen only for one particular family.
-
-An academic paper accompanies work on this tool, describing, for example, motivation of design choices. 
-It will be published here after peer-review process.
-## The idea
-Basic assumption of this project is that HTTP requests of different malware families are more or less unique, 
-so they can be fingerprinted to provide some sort of identification. Hfinger retains information about structure and 
+It's main objective is to provide unique representations (fingerprints) of malware requests,
```
Comment (nazywam) [inline README.md]:
```suggestion
Its main objective is to provide unique representations (fingerprints) of malware requests, 
```

File: README.md [hunk 3]
```diff
@@ -1,108 +1,180 @@
...
+For example:
 
+```console
+foo@bar:~$ hfinger -f /tmp/test.pcap
+[{"epoch_time": "1614098832.205385000", "ip_src": "127.0.0.1", "ip_dst": "127.0.0.1", "port_src": "53664", "port_dst": "8080", "fingerprint": "2|3|1|php|0.6|PO|1|us-ag,ac,ac-en,ho,co,co-ty,co-le|us-ag:f452d7a9/ac:as-as/ac-en:id/co:Ke-Al/co-ty:te-pl|A|4|1.4"}]
+```
 
-## Usage
-Calling the tool from a command line:
+Help is displayed with short `-h` or long `--help` switches:
```
Comment (nazywam) [inline README.md]:
```suggestion
Help can be displayed with short `-h` or long `--help` switches:
```

File: hfinger/analysis.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        try:
+            filelist = os.listdir(args.directory)
+        except NotADirectoryError as err:
+            print("The entered path is not a directory. " + str(err))
+            sys.exit(1)
+        for x in os.listdir(args.directory):
+            cur_file = os.path.join(args.directory, x)
+            try:
+                is_pcap_file(cur_file)
+            except IsADirectoryError:
```
Comment (nazywam) [inline hfinger/analysis.py:140]: IMO checking it using `os.is_file` could be a bit more elegant than catching exceptions

File: setup.py (hunk 1 above)
Comment (PBia) [reply setup.py]: Deleted in https://github.com/CERT-Polska/hfinger/commit/e5b2758b5a67f3ff619a9de2fa72e6408b5d35f8

File: README.md (hunk 2 above)
Comment (PBia) [reply README.md]: Changed in https://github.com/CERT-Polska/hfinger/commit/e5b2758b5a67f3ff619a9de2fa72e6408b5d35f8

File: README.md (hunk 3 above)
Comment (PBia) [reply README.md]: Changed in https://github.com/CERT-Polska/hfinger/commit/e5b2758b5a67f3ff619a9de2fa72e6408b5d35f8

**Review by chivay — APPROVED** [review]

**Pull request merged**